### DISCUSSION TITLE
Help：agent not support blocking mode in restful api，why?

### DISCUSSION THREAD

**Discussion by zhukefucn** [langgenius/dify#7805] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

My program uses restful api to call agent in dify application service , mode = streaming is ok ,but not support blocking mode ,why?
it shows " Agent Chat App does not support blocking mode"

**Comment by liroys27** [comment]
I have the same issue, blocking mode not supported by agent

**Comment by Daishuyuan** [comment]
I have the same issue. When I use cursor to call dify mcp server, it throws 'agent chat app does not support blocking mode'.
<img width="390" alt="截屏2025-04-01 18 51 19" src="https://github.com/user-attachments/assets/6291eecf-2d50-479d-8084-0760038ebb02" />
my cursor mcp.json follows:
```json
{
  "mcpServers": {
    "DataTableDesigner": {
      "url": "xxx/sse",
      "proxy": false,
      "strictSSL": false,
      "timeout": 120000,
      "retryAttempts": 5,
      "retryDelay": 3000,
      "keepAlive": true,
      "reconnect": true,
      "reconnectDelay": 1000,
      "maxReconnectAttempts": 10,
      "heartbeatInterval": 30000,
      "nonBlocking": true
    }
  }
}
```
by the way, I use the extension: mcp-server.
